### ISSUE TITLE
`display.clear()` does not stop `display.show(loop=True, wait=False)`

### ISSUE THREAD

**Issue by microbit-carlos** [microbit-foundation/micropython-microbit-v2#150]
Reported in the V1 repository, but I think it's actually a V2 only issue:
- https://github.com/bbcmicrobit/micropython/issues/789

With this programme pressing button A should stop the animation looping forever by using `display.clear()`, but it only clears the current image displayed and the next animation steps continue.

Pressing B replaces the animation by just displaying a letter and at that point `display.clear()` works as expected.

```python
from microbit import *

display.show("hello", loop=True, wait=False)
while True:
    sleep(100)
    if button_a.is_pressed():
        display.clear()
    if button_b.is_pressed():
        display.show("X")
```

In V1 `display.clear()` does stop the animation.

**Comment by davidtkaye** [comment]
Can confirm this is a problem on my V2 microbit, also coincidentally shows up on the microbit simulator in the online Python editor (where I first identified it).

**Comment by dpgeorge** [comment]
Thanks for the report.  Should be fixed by a5dd649283ad6c232e5f924af3b24242f58e3bbc

**Comment by dpgeorge** [comment]
@microbit-matt-hillsdon for the simulator you can now remove `microbit_hal_display_clear()`, per 5177bdc96df45266f344b2813f2f4d94891c7928
